### DISCUSSION TITLE
Genres not pulling correctly

### DISCUSSION THREAD

**Discussion by andithelibrarian** [seanap/Plex-Audiobook-Guide#110] [General]
The genres seem to be pulling from Category and not the actual genre. As a result instead of genres like Horror or Mystery, I am often getting things like Teen & Young Adult or Literature.

**Comment by seanap** [comment]
Please provide an example. We are limited to what audible provides, but it should include all genres and sub genres provided.

**Reply by andithelibrarian** [reply]
Ok. So here's and example. This book, The Amish Matchmakers, pulls Literature & Fiction as the Genre. It's getting this from the Category. But as you can see in the screencap below, the information below it, is more properly the genres, particularly Romance and Funny.

In general, I find that Audible's tags have better genre information than their Categories. But Libro.fm is even better most of the time. I don't know if it's even possible to make an option to pull from them, but it might be worth looking into as an option.

<img width="287" alt="Screenshot 2024-05-15 at 6 47 36 PM" src="https://github.com/seanap/Plex-Audiobook-Guide/assets/169846772/2ef7d152-c85b-4a5f-8781-25a5f57a9804">
<img width="1268" alt="Screenshot 2024-05-15 at 6 45 10 PM" src="https://github.com/seanap/Plex-Audiobook-Guide/assets/169846772/bde83c53-2955-4df2-9d1e-fd4314799a42">

**Reply by seanap** [reply]
It looks like audible is now using those Chips for sub-genres.  The script is scraping the Categories for the genre and it looks like that is still working just fine.  They used to have multiple categories and I simply took the first two categories and combined them like "Literature & Fiction/Amish & Mennonite"

It is possible to append the chips to the category, eg: "Literature & Fiction/Amish & Mennonite/Romance/Clean & Wholesome/Funny/Sibling/Hotel" but that doesn't look too helpful for me to have embedded in the file.

The issue is the ID3 tags really arn't designed for multiple genres which gives us that ugly long tag. We have no control over which sub-genres audible puts as chips, nor can we reorder them, and my script isn't smart enough to only use "relevant" genres like Romance and Fiction.

Mp3tag source scripts can only connect to a single source, in this case Audible book details page.  To scrape libro.fm you would need to write a completely new scraper to scrape that site.

Another thing I find really interesting is that Audible changed their html code and left "seriesLabel" in the source code even if it's not a series.  I'll have to fix the script so it checks to see if there's a value in that field so it doesn't get all wonky.

**Reply by andithelibrarian** [reply]
Thanks. It does look like on a lot of the Audible they do have too many chips to be useful. It's a shame they seem to have moved the major genres down to the chips along with the other tags. I did a search and it does look like someone has made a scraper for Libro.fm and it's been updated in the last year. I may add it and see how it does. I won't post the link here now in case that's not cool with you. But let me know if it's ok and I'll come back and post the link once I've tried it to see how well it works. Thanks again! I'm getting ready to make the switch to audiobookshelf, but this works for that too, and it saves a ton of time.
